### ISSUE TITLE
installation of SZ

### ISSUE THREAD

**Issue by lenalana** [szcompressor/SZ#83]
hello, I'm new in linux and i'm very confused. 
I installing sz by using this link 
 git clone https://github.com/disheng222/SZ
and now iam trying to go to step 2 

> mkdir build && cd build
> cmake .. -DCMAKE_INSTALL_PREFIX:PATH=[INSTALL_DIR]
> make
> make install

I creat build directory and cd build  and then 

> /lin/build$ cmake -- -DCMAKE_INSTALL_PREFIX:PATH=/home/mainuu/lin/build

I get on this error ?? how can fix it please ?

> CMake Error: The source directory "/home/mainuu/lin/build/--" does not exist.
> Specify --help for usage, or press the help button on the CMake GUI.
>

**Comment by robertu94** [comment]
@lenalana welcome to Linux and thanks for trying SZ. 

CMake requires a build directory the ‘.’ (Period) in the command from the readme.  I didn’t see it in the command you copied above. Additionally there is no space between — and DCMAKE_INSTALL_PREFIX:PATH.  Finally you have an extra ‘-‘ in the command you posted. Hope this helps!

**Comment by lenalana** [comment]
The file CMakeLists.txt  and the other files of sz in the build directory , Is that ok ? and I used 
/lin/build$ cmake --DCMAKE_INSTALL_PREFIX:PATH=/home/mainuu/lin/build
and the same error?

**Comment by robertu94** [comment]
tl;dr I would delete your clone and start over by copying and pasting the code below into your terminal.  You may also want to copy the first 2 lines into your `$HOME/.bashrc` file.

The `CMAKE_INSTALL_PREFIX` path is where cmake will install files to.  You shouldn't use the `build` directory because CMake can and will overwrite files in that directory at any time.  It should point to a location that is on your `PATH` variable in your shell (`echo $PATH` to see it).  For example I use this set of command when I want to build SZ personally.

```bash
export PATH=$HOME/.local/bin:$PATH
export LD_LIBRARY_PATH=$HOME/.local/lib:$HOME/.local/lib64:$LD_LIBRARY_PATH
git clone https://github.com/szcompressor/SZ
cd SZ
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/.local
make
make install
```

Now let me explain what this does.  The first 2 lines add the prefix `$HOME/.local` so that your shell (the terminal app) knows where to find them. A prefix is just a tree of related software.  It is a phrase that dates back to GNU Autotools (one of the first build systems). `PATH` is for executables and `LD_LIBRARY_PATH` is for libraries (on linux these tend to start with the prefix `lib` and to end with the extension `.so` or `.a`).  We extend their values by saying `a=new_item:$a` because we want to extend the list by prepending to it so it is searched first. Calling `export` then allows programs called from the shell to see these variables too rather than just the shell.

Next we clone the repository from the internet using git. And make the build directories.

Lastly we invoke cmake to generate `Makefile`s, and then use `make` to run the actual build. Depending on your version of cmake there is a slightly easier way to do this, but the commands I provided here should work on any version of cmake from the last 5 or so years.

**Comment by lenalana** [comment]
Thank you so much , I get on this :) 
:~/SZ/build$ cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/.local
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.10.2 or higher is required.  You are running version 3.7.2

**Comment by epasveer** [comment]
You need to update the version of cmake that you are using.   Which linux distribution are you using?  Most distributions have the recent cmake.

**Comment by disheng222** [comment]
Please note that cmake is followed by ".." instead of "--

Specially, cmake -DCMAKE_INSTALL_PREFIX:PATH=/home/mainuu/lin/build ..


Best,
Sheng

**Comment by lenalana** [comment]
hello again, I'm trying to execute this command but I get this :
```
/SZ/example$ ./testdouble_compress.c  sz.config testdata/x86/testdouble_8_8_128.dat 8 8 128
bash: ./testdouble_compress.c: Permission denied
```

and with this 
`~/SZ/example$ ./test.sh `
============== testing compression and decompression of 1D array ==============
./testdouble_compress sz.config testdata/x86/testdouble_8_8_128.dat 8192
./test.sh: line 9: ./testdouble_compress: No such file or directory
./testdouble_decompress testdata/x86/testdouble_8_8_128.dat.sz 8192
./test.sh: line 11: ./testdouble_decompress: No such file or directory
./testfloat_compress sz.config testdata/x86/testfloat_8_8_128.dat 8192
./test.sh: line 14: ./testfloat_compress: No such file or directory
./testfloat_decompress testdata/x86/testfloat_8_8_128.dat.sz 8192
./test.sh: line 16: ./testfloat_decompress: No such file or directory

**Comment by disheng222** [comment]
You need to compile ./testdouble_compress.c to get the executable file:
./testdouble_compress.
The command should be like "./testdouble_compress sz.config
testdata/x86/testdouble_8_8_128.dat 8 8 128"

I suggest you using the command 'sz' instead. You also need to compile sz.c
to get the executable file -- sz.

**Comment by lenalana** [comment]
Hi disheng ,I'm very confused now. 
now I compile the  file by use :
:~/SZ/example$ gcc -o testsd testdouble_compress.c 
testdouble_compress.c:13:16: fatal error: sz.h: No such file or directory
 #include "sz.h"
                ^
compilation terminated.

What is the correct path that I should use pleace ?

**Comment by disheng222** [comment]
Hi Iana,
I suggest you read the README carefully.
I copied the related parts as follows:

./configure --prefix=[INSTALL_DIR] (Please use --enable-fortran if you need
Fortran interface)
make
make install

You should replace [INSTALL_DIR] by a real path on your machine. In this
way, the executables will be copied in the [INSTALL_DIR].

For simplicity, you can just run the following commands to compile the
code:

cd SZ
./configure
make

(Then, you will find the compiled executables in ./example/ directory.)

Best,
Sheng

**Comment by lenalana** [comment]
Actually, I've read it, but I'm having trouble with Linux commands.
I have done the following 

```
:~/SZ$ make
make  all-recursive
make[1]: Entering directory '/home/mainuu/SZ'
Making all in zlib
make[2]: Entering directory '/home/mainuu/SZ/zlib'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/mainuu/SZ/zlib'
Making all in zstd
make[2]: Entering directory '/home/mainuu/SZ/zstd'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/mainuu/SZ/zstd'
Making all in sz
make[2]: Entering directory '/home/mainuu/SZ/sz'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/mainuu/SZ/sz'
Making all in example
make[2]: Entering directory '/home/mainuu/SZ/example'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/mainuu/SZ/example'
make[2]: Entering directory '/home/mainuu/SZ'
make[2]: Leaving directory '/home/mainuu/SZ'
make[1]: Leaving directory '/home/mainuu/SZ'
```

```
~/SZ$ cd example 
:~/SZ/example$ gcc -o testsd testdouble_compress.c
testdouble_compress.c:13:16: fatal error: sz.h: No such file or directory
 #include "sz.h"
                ^
compilation terminated.
```
Do you need to transfer these files like "sz.h" and other to the same place? Or?

**Comment by lenalana** [comment]
What about Run ? 
~/SZ/example$ sz -z -d -c sz.config -i testdata/x86/testdouble_8_8_128.dat -3 8 8 128 -a 
bash: sz: command not found

**Comment by lenalana** [comment]
Thank you so much, final question :) 
if I need to create my program in another directory like(newdir) , what I need to be in side this newdir? because I run /configure inside it but get on (bash: ./configure: No such file or directory)

**Comment by disheng222** [comment]
Under ~/SZ/example, you need to run ./sz instead of sz, or you can use the
full path ~/SZ/example/sz to run that command.
This is a basic knowledge about Linux commands. I suggest you learn Linux
beforehand.

**Comment by disheng222** [comment]
You can execute the following to generate the executables and library:
./configure --prefix=[install_dir]
make
make install

[install_dir] is the target dir you want to create and copy everything
compiled into it.
I suggest you learn automake or Linux beforehand.

Best,
Sheng
